components/datetime/src/provider/calendar/mod.rs

```diff
@@ -15,19 +15,23 @@ use icu_provider::{yoke, zerofrom};
 pub use skeletons::*;
 pub use symbols::*;
 
-#[icu_provider::data_struct(marker(
-    DateLengthsV1Marker,
-    "datetime/datelengths@1",
-    extension_key = "ca"
-))]
+#[icu_provider::data_struct(
+    marker(GregorianDateLengthsV1Marker, "datetime/gregory/datelengths@1"),
```

Comment:
thought: we can use this opportunity to call it datetime/gregorian, perhaps? Ideally we don't have to have the "gregory" name anywhere that is not BCP-47 constrained

Reply:
I think I'd rather stick to the official identifiers. There is "gregorian" in CLDR JSON but that's not really standard. It opens questions about whether ethioaa, japanext, etc should be expanded, too